src/syft/core/node/common/plan/plan.py

```diff
@@ -0,0 +1,140 @@
+# stdlib
+import re
+import sys
+from typing import Any
+from typing import List
+from typing import Tuple
+from typing import Union
+
+# third party
+from google.protobuf.reflection import GeneratedProtocolMessageType
+from nacl.signing import VerifyKey
+
+# syft absolute
+from syft.core.common.object import Serializable
+from syft.core.node.common.action.common import Action
+from syft.core.pointer.pointer import Pointer
+from syft.proto.core.node.common.action.action_pb2 import Action as Action_PB
+from syft.proto.core.node.common.plan.plan_pb2 import Plan as Plan_PB
+
+# syft relative
+from ...abstract.node import AbstractNode
+from ..util import listify
+
+CAMEL_TO_SNAKE_PAT = re.compile(r"(?<!^)(?=[A-Z])")
+
+
+class Plan(Serializable):
+    """
+    A plan is a collection of actions, plus some variable inputs, that together form a computation graph.
+
+    Attributes:
+        actions: list of actions
+        inputs: Pointers to the inputs. Defaults to None.
+    """
+
+    def __init__(
+        self, actions: List[Action], inputs: Union[Pointer, List[Pointer], None] = None
+    ):
+        self.actions = actions
+        self.inputs: List[Pointer] = listify(inputs)
+
+    def execute(
```

Comment:
Let's get some more documentation inline in this guy

Reply:
✔️

Reply:
Could we make this `__call__` as well?